docs/source/developer_guide/architecture.md

```diff
@@ -55,7 +55,7 @@ The stage identifier is a unique string used in both logging and creating the st
 ### Type Inference
 To perform work, each stage needs to know what type of data it will be operating on. Since Morpheus can pass any type of data from stage to stage, the pipeline must ensure compatible types at every edge connection between stages. This process is called stage type inference and is performed during the pipeline build phase.
 
-Stage type inference is necessary because the output type of some stages may depend on the output type of the previous stage. For example, consider a simple pass through stage that passes the input message to the next stage unmodified. If our pass through stage is connected to another stage generating a string, its output type will be a string. Instead, if it's connected to another stage generating an integer, its output type will be an integer.
+Stage type inference is necessary because the output type of some stages may depend on the output type of the previous stage. For example, consider a simple pass-through stage that passes the input message to the next stage unmodified. If our pass-through stage is preceded by a stage generating a string, its output type will be a string. Instead, if it's preceded by a stage generating an integer, its output type will be an integer.
```

Comment:
Do you recall if the editor said always use two words or was there an exception for being used as an adjective? I thought "pass-through X" was hyphenated because it's an adjective modifying X. Happy to change it either way just let me know :-)

Reply:
If I recall correctly the importance was on being consistent with the choice. In an early rough draft I had used all three variants of "pass through".  So mostly she just wanted us to pick one and stick with it.

Reply:
ok I will revert to "pass through" and suggest changes; please resolve/commit as you see fit

Reply:
```suggestion
Stage type inference is necessary because the output type of some stages may depend on the output type of the previous stage. For example, consider a simple pass through stage that passes the input message to the next stage unmodified. If our pass through stage is preceded by a stage generating a string, its output type will be a string. Instead, if it's preceded by a stage generating an integer, its output type will be an integer.
```